worker/internal/benthos/transformers/utils/string_utils.go

```diff
@@ -134,11 +134,12 @@ func IsValidChar(s string) bool {
 }
 
 func IsAllowedSpecialChar(r rune) bool {
-	allowedSpecialChars := "!@#$%^&*()-+=_[]{}|\\ ;\"<>,./?"
-	for _, allowedChar := range allowedSpecialChars {
-		if r == allowedChar {
-			return true
-		}
+	allowedSpecialChars := map[rune]struct{}{
```

Comment:
It's beneficial to move stuff like this outside of the function so that it doesn't get re-defined every time the function is invoked. 

If it's moved to be defined at the module level, then it's defined and allocated once by the process, and then never again. Where as the way it currently is, the process has to define this map, check if the rune is in it, then throw it away. 

Should look something like this:

```
var (
  allowedSpecialChars = map[rune]struct{}{...}
)

func IsAllowedSpecialChar(r rune) bool {
  _, ok := allowedSpecialChars[r]
  return ok
}
```

Reply:
makes sense, updated